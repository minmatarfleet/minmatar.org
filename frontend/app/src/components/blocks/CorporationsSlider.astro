---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { CorporationObject } from '@dtypes/layout_components'

interface Props {
    corporations?:          CorporationObject[];
    user_corporation_id?:   number;
}

let {
    corporations = [],
    user_corporation_id = 0,
} = Astro.props

const CORPORATIONS_WIDTH_SLIDE = [
    'Rattini Tribe',
    'Black Omega Security',
    'Balls Deep Inc.',
    'Vindicta Expeditionary',
    'Minmatar Fleet Academy',
    'Rapid Unscheduled Disassembly.',
    'Soltech Armada',
    'Exit-Strategy',
]

let corporations_with_no_slide:CorporationObject[] = []
let corp_filter:number[] = []

import { get_corporation_logo } from '@helpers/eve_image_server';
import { get_alliance_logo } from '@helpers/eve_image_server';

corporations = corporations.filter( (i) => i.corporation_id != 98735318 ) // Hide Minmatar Fleet Holdings

const pagination_icons:string[] = []
corporations.forEach( (corp, i) => {
    if (!CORPORATIONS_WIDTH_SLIDE.includes(corp.corporation_name)) {
        corporations_with_no_slide.push(corp)
        corp_filter.push(corp.corporation_id)
    }
});

corporations = corporations.filter( (i) => !corp_filter.includes(i.corporation_id) )
let corporations_sorted:CorporationObject[] = []

CORPORATIONS_WIDTH_SLIDE.forEach(corp_name => {
    const corp = corporations.find(corporation => corporation.corporation_name === corp_name)
    if (corp !== undefined) {
        corporations_sorted.push(corp)
        pagination_icons.push(get_corporation_logo(corp.corporation_id, 64))
    }
})
              
const MINMATAR_FLEET_ID = 99011978
const MINMATAR_FLEET_ASSOCIATES_ID = 99012009

if (corporations_with_no_slide.length > 0)
    pagination_icons.push(get_alliance_logo(MINMATAR_FLEET_ID, 64))

pagination_icons.push(get_alliance_logo(MINMATAR_FLEET_ASSOCIATES_ID, 64))

import { slugify } from '@helpers/string'

import Wrapper from '@components/compositions/Wrapper.astro';

import CorporationsSwiper from '@components/blocks/CorporationsSwiper.astro';
import RattiniSlide from '@components/blocks/RattiniSlide.astro';
import BlackOmegaSlide from '@components/blocks/BlackOmegaSlide.astro';
import BallsDeepSlide from '@components/blocks/BallsDeepSlide.astro';
import VindictaExpeditionarySlider from '@components/blocks/VindictaExpeditionarySlider.astro';
import RapidUnscheduledDisassemblySlide from '@components/blocks/RapidUnscheduledDisassemblySlide.astro';
import L34RnSlide from '@components/blocks/L34rnSlide.astro';
import SoltechArmadaSlide from '@components/blocks/SoltechArmadaSlide.astro';
import ExitStrategySlide from '@components/blocks/ExitStrategySlide.astro';
import AssociatesSlide from '@components/blocks/AssociatesSlide.astro';
import FleetCorporationsSlide from '@components/blocks/FleetCorporationsSlide.astro';
---

<div
    class="[ swiper ][ w-full ]"
    class:list={[ 'w-full', { swiper: corporations.length > 0 } ]}
    id="corporations-list"
>
    {corporations_sorted.length > 0 &&
        <CorporationsSwiper
            pagination_icons={pagination_icons}
        >
            {corporations_sorted.map((corporation, index) =>
                <div class="swiper-slide" data-hash={slugify(corporation.corporation_name)}>
                    {corporation.corporation_name == 'Rattini Tribe' &&
                        <RattiniSlide
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                    {corporation.corporation_name == 'Black Omega Security' &&
                        <BlackOmegaSlide
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                    {corporation.corporation_name == 'Balls Deep Inc.' &&
                        <BallsDeepSlide
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                    {corporation.corporation_name == 'Vindicta Expeditionary' &&
                        <VindictaExpeditionarySlider
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                    {corporation.corporation_name == 'Minmatar Fleet Academy' &&
                        <L34RnSlide
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                    {corporation.corporation_name == 'Rapid Unscheduled Disassembly.' &&
                        <RapidUnscheduledDisassemblySlide
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                    {corporation.corporation_name == 'Soltech Armada' &&
                        <SoltechArmadaSlide
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                    {corporation.corporation_name == 'Exit-Strategy' &&
                        <ExitStrategySlide
                            corporation={corporation}
                            is_user_corporation={corporation.corporation_id === user_corporation_id}
                            picture_placement={index%2 == 0 ? 'left' : 'right'}
                        />
                    }
                </div>
            )}
            {corporations_with_no_slide.length > 0 &&
                <div class="swiper-slide" data-hash="Minmatar_Fleet_New_Corporations">
                    <FleetCorporationsSlide
                        id="slideless-alliance-corps"
                        picture_placement={corporations.length%2 == 0 ? 'left' : 'right'}
                        user_corporation_id={user_corporation_id}
                        corporations={corporations_with_no_slide}
                    />
                </div>
            }
            <div class="swiper-slide" data-hash="Minmatar_Fleet_Associates">
                <AssociatesSlide
                    id="associates-alliance"
                    picture_placement={(corporations.length + (corporations_with_no_slide.length > 0 ? 1 : 0))%2 == 0 ? 'left' : 'right'}
                />
            </div>
        </CorporationsSwiper>
    }
    {corporations.length == 0 &&
        <Wrapper padding_inline='0'>
            <slot>
                {t('empty_corporations_text')}
            </slot>
        </Wrapper>
    }
</div>