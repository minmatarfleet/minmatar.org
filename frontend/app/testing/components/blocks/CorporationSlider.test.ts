import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CorporationsSlider from "@components/blocks/CorporationsSlider.astro";

test("CorporationsSlider defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(CorporationsSlider, {
    props: {
      corporations: [],
    },
    slots: {
      default: "slotted chick",
    },
  });

  // expect(result).toMatchSnapshot();
});
