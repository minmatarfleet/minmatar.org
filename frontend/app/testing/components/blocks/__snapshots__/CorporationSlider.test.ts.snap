// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorporationsSlider defaults 1`] = `"<div class="[ swiper ][ w-full ] w-full" id="corporations-list">  <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: var(--space-2xl);--wrapper_max_width: 80rem;">  slotted chick  </div>   </div>"`;
